newIDE/app/src/AssetStore/NewObjectDialog.js

```diff
@@ -198,9 +204,21 @@ export default function NewObjectDialog({
       events,
       onObjectAddedFromAsset,
       openedAssetShortHeader,
+      environment,
     ]
   );
 
+  // Load assets and filters when the dialog is opened.
+  React.useEffect(
+    () => {
+      // This check is to ensure we call fetchAssetsAndFilters every time environment changes.
+      if (environment) {
```

Comment:
environment can't be null right?

Reply:
No, it's a bit of hack, but couldn't find another way to trigger a refetch when the value change 🤔 
You have to do it inside an effect, because if you do it just after the value is set in the state, it's not there yet (and no callback available)
I can improve the comment to indicate why it's a hack